More consistent Docker unit parsing

Christoph-D opened the pull request:
##### ISSUE TYPE
- Bugfix Pull Request
##### SUMMARY

Issue #16526 shows that there is confusion about the valid units for memory sizes in docker.  This change adds `kernel_memory` to the list of parameters whose value needs to go through `human_to_bytes()` and updates the documentation.

calfonso commented:
@Christoph-D This needs to be rebased, then use the basic.py functions. Once that is done and you've verified everything, please close this PR, thanks!

Christoph-D committed:
Docker: Convert units for kernel_memory

Other memory arguments already use human_to_bytes to convert the
units.  This change makes the behavior more consistent.

Christoph-D commented:
I rebased the PR, it is now much easier and mostly a documentation update. PTAL?

felixfontein commented:
@Christoph-D I just found your PR; I did something very similar three weeks ago: #42789

Both our PRs fix #42692, so definitely one of them should be merged. I think yours is better, as it also documents some more valid units :) Except that you should not use backticks, you should use `C(...)` instead to get the documentation formatted correctly.

@kassiansun Can you also take a look? You seem to be the only active `docker_container` maintainer currently...

felixfontein commented on lib/ansible/modules/cloud/docker/docker_container.py:
```diff
@@ -166,7 +166,9 @@
     description:
       - "Kernel memory limit (format: <number>[<unit>]). Number is a positive integer.
-        Unit can be one of b, k, m, or g. Minimum is 4M."
+        Unit can be `B` (byte), `K` (kibibyte, 1024B), `M` (mebibyte), `G` (gibibyte),
```
Please use `C(...)` instead of backticks.

felixfontein commented on lib/ansible/modules/cloud/docker/docker_container.py:
```diff
@@ -199,17 +201,23 @@
       - "Memory soft limit (format: <number>[<unit>]). Number is a positive integer.
-        Unit can be one of b, k, m, or g"
+        Unit can be `B` (byte), `K` (kibibyte, 1024B), `M` (mebibyte), `G` (gibibyte),
+        `T` (tebibyte), or `P` (pebibyte)."
```
Same here.

felixfontein commented on lib/ansible/modules/cloud/docker/docker_container.py:
```diff
@@ -199,17 +201,23 @@
-        Number is a positive integer. Unit can be one of b, k, m, or g.
+      - "Total memory limit (memory + swap, format:<number>[<unit>]).
+        Number is a positive integer. Unit can be `B` (byte), `K` (kibibyte, 1024B),
+        `M` (mebibyte), `G` (gibibyte), `T` (tebibyte), or `P` (pebibyte)."
```
Same here.

felixfontein commented on lib/ansible/modules/cloud/docker/docker_container.py:
```diff
@@ -322,9 +330,10 @@
-      - Omitting the unit defaults to bytes. If you omit the size entirely, the system uses `64m`.
+      - "Size of `/dev/shm` (format: `<number>[<unit>]`). Number is positive integer.
+        Unit can be `B` (byte), `K` (kibibyte, 1024B), `M` (mebibyte), `G` (gibibyte),
+        `T` (tebibyte), or `P` (pebibyte)."
```
Same here.

felixfontein commented on lib/ansible/modules/cloud/docker/docker_container.py:
```diff
@@ -166,7 +166,9 @@
   kernel_memory:
     description:
       - "Kernel memory limit (format: <number>[<unit>]). Number is a positive integer.
```
`<number>[<unit>]` should be formatted with `C(...)` as well (everywhere). Currently, backticks are only used for `shm_size` (which should be converted to `C(...)` too).

felixfontein reviewed (changes requested):
Except some minimal changes (backticks -> `C(...)` and some more unification), this is great.

Christoph-D commented:
I changed the backticks to C(...).

Christoph-D committed:
Update Docker unit parsing documentation

This explains the units more precisely and makes the documentation
more consistent.

This fixes #16526.

felixfontein reviewed (approved):
Assuming you removed the executable attribute, looks good to me! :)

shipit

felixfontein commented:
bot_status

felixfontein commented:
@kassiansun I guess you have to re-shipit this one.

kassiansun commented:
shipit

felixfontein commented:
bot_status

felixfontein commented:
shipit

felixfontein commented:
bot_status

felixfontein commented:
yay, progress :tada:
